autoload/utils/exec/job.vim

```diff
@@ -21,21 +21,30 @@ function! s:closeBuffer() abort
 endfunction
 
 function! s:createQuickFix() abort
+    " just to be sure all messages were processed
+    sleep 100m
     let l:bufnr = bufnr(s:cmake4vim_buf)
     if l:bufnr == -1
         return
     endif
     let l:old_error = &errorformat
-    if s:err_fmt !=# ''
-        let &errorformat = s:err_fmt
+    if s:cmake4vim_job['err_fmt'] !=# ''
+        let &errorformat = s:cmake4vim_job['err_fmt']
     endif
+
     silent execute 'cgetbuffer ' . l:bufnr
     silent call setqflist( [], 'a', { 'title' : s:cmake4vim_job[ 'cmd' ] } )
-    if s:err_fmt !=# ''
+    if s:cmake4vim_job['err_fmt'] !=# ''
         let &errorformat = l:old_error
     endif
     " Remove cmake4vim job
     let s:cmake4vim_job = {}
+    call s:closeBuffer()
+    if len(s:cmake4vim_jobs_pool) > 0
+        let l:next_job = s:cmake4vim_jobs_pool[0]
```

Comment:
I think this can be written as:
```suggestion
if !empty( s:cmake4vim_jobs_pool)
        let [ l:next_job; s:cmake4vim_jobs_pool ] = s:cmake4vim_jobs_pool        
        silent call utils#exec#job#run(l:next_job['cmd'], l:next_job['open_qf'], l:next_job['err_fmt'])
endif
```

Check `:h let-unpack`
```
:let [{name}, ..., ; {lastname}] = {expr1}				*E452*
			Like |:let-unpack| above, but the |List| may have more
			items than there are names.  A list of the remaining
			items is assigned to {lastname}.  If there are no
			remaining items {lastname} is set to an empty list.
			Example: >
				:let [a, b; rest] = ["aval", "bval", 3, 4]
```

Reply:
Good advice.